include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,760 @@
+        assign(_lit);
+        return *this;
+    }
+
+    template <size_t N>
+    constexpr void assign(char const (&_lit)[N])
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of dynamic_bitset.");
+        assert(_lit[N - 1] == '\0');
+        uint64_t value{};
+        for (size_t i = 0; i != N - 1; ++i)
+        {
+            if (_lit[i] == '0')
+            {
+                value <<= 1;
+            }
+            else if (_lit[i] == '1')
+            {
+                if (i != 0)
+                {
+                    value <<= 1;
+                }
+                value |= 1u;
+            }
+            else
+            {
+                throw std::invalid_argument{"Only 1 or 0."};
+            }
+        }
+
+        (*this) = value;
+    }
+
+    constexpr void assign(std::initializer_list<value_type> ilist) noexcept
+    {
+        assign(std::ranges::begin(ilist), std::ranges::end(ilist));
+    }
+
+    constexpr void assign(size_type const count, value_type const value) noexcept
+    {
+        clear();
+        auto tmp = view::repeat_n(value, count);
+        assign(std::ranges::begin(tmp), std::ranges::end(tmp));
+    }
+
+    template <std::ranges::InputRange other_range_t>
+    //!\cond
+        requires std::Constructible<value_type, /*ranges::range_reference_t*/reference_t<other_range_t>>
+    //!\endcond
+    constexpr void assign(other_range_t && range)
+    {
+        assign(std::ranges::begin(range), std::ranges::end(range));
+    }
+
+    template <std::ForwardIterator begin_it_type, std::Sentinel<begin_it_type> end_it_type>
+    constexpr void assign(begin_it_type begin_it, end_it_type end_it) noexcept
+    //!\cond
+        requires std::Constructible<value_type, /*ranges::iter_reference_t*/reference_t<begin_it_type>>
+    //!\endcond
+    {
+        clear();
+        insert(cbegin(), begin_it, end_it);
+    }
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    //!\brief Returns the begin to the string.
+    constexpr iterator begin() noexcept
+    {
+        return iterator{*this};
+    }
+
+    //!\copydoc begin()
+    constexpr const_iterator begin() const noexcept
+    {
+        return const_iterator{*this};
+    }
+
+    //!\copydoc begin()
+    constexpr const_iterator cbegin() const noexcept
+    {
+        return const_iterator{*this};
+    }
+
+    //!\brief Returns iterator past the end of the vector.
+    constexpr iterator end() noexcept
+    {
+        return iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    constexpr const_iterator end() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    constexpr const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+    //!\}
+
+    constexpr bool all() const noexcept
+    {
+        return !size() == 0 && count() == size();
```

Comment:
`!size() == 0` ? this seems wierd

Reply:
`all()` has to return `false` for an empty `std::bitset`

Reply:
why? normally it returns the neutral element and this is true for empty

Reply:
https://en.cppreference.com/w/cpp/utility/bitset/all_any_none

🤷‍♂

Reply:
Even though `std::ranges::all_of` and `std::all_of` return `true` for empty ranges.

Reply:
> true if all bits are set to true, otherwise false

This is now interpretation, but I would have said that all (0-many) bits are set true :joy:

Reply:
Would be more consistent and intuitive, and easier to implement :)

Reply:
https://godbolt.org/z/8IqQSj

Says that empty many bits are true for all.

Reply:
You're right. It has to be `std::bitset<0>` for the std one 😅